### ISSUE TITLE
[BUG]: `sizeof(uintptr_t) != sizeof(void *)` in MIPS 64 clang

### ISSUE THREAD

**Issue by firejox** [compiler-explorer/compiler-explorer#5003]
### Describe the bug

Move the issue from https://github.com/llvm/llvm-project/issues/62398. The size of `uintptr_t` will be smaller than the size of pointer in MIPS-64-clang, but MIPS-64-gcc won't. 

### Steps to reproduce

1. Paste the following code in godbolt.org
```c
#include <stdint.h>
#include <stddef.h>

void check(void) {
    uintptr_t v = (uintptr_t)NULL;
    _Static_assert(sizeof(uintptr_t) == sizeof(void *), "It should be same size");
}
```
2. Choose the compiler with MIPS 64 clang
3. You will see the warning and static assert failed.

### Expected behavior

No Warn, No Assertion Failed

### Reproduction link

https://godbolt.org/z/8G4dWcv9j

### Screenshots

![image](https://user-images.githubusercontent.com/6823999/234931776-0293d23a-d9ea-4df6-a1cd-cbe317c71251.png)

**Comment by firejox** [comment]
@dkm I found that it is similar to #4156. Using `--gcc-toolchain` would resolve the problem.
https://godbolt.org/z/bKKEr1Ker

**Comment by dkm** [comment]
Yes, that's what I was thinking... IIRC, I'm the one who added thess mips clangs config...

Thanks for confirming, will make the fix easier :)
